libs/wingc/src/lib.rs

```diff
@@ -272,6 +291,8 @@ pub fn compile(
 		&mut asts,
 	);
 
+	emit_warning_for_unsupported_package_managers(&project_dir);
```

Comment:
To avoid the cost of multiple syscalls for the average case, what if this check only happens if there are errors? Ideally, we'd only want to check this if we specifically fail package resolution.

Reply:
> Ideally, we'd only want to check this if we specifically fail package resolution.

Perhaps. But even if you use pnpm initially and only add a winglib later, getting the warning earlier in your journey as a first-time user might be better, I'd argue.

For now I'm optimizing for the pain of users being unable to compile their Wing apps, since that's a critical point where users can drop off (they might get confused by the error, have no idea how to fix it, and give up before asking on Discord or GitHub). Minimizing the number of warnings for happy Wing users who insist on using a different package manager is a secondary issue.